### DISCUSSION TITLE
Add isToggledHiddenByDefault() Condition to assertTableColumnHidden Testing Function

### DISCUSSION THREAD

**Discussion by misterneo** [filamentphp/filament#15214] [Ideas]
Why is the isToggledHiddenByDefault() condition not included in the assertTableColumnHidden() testing function?

![Screenshot 2024-12-31 at 14 34 47](https://github.com/user-attachments/assets/8e3ebea7-5769-48f1-9733-5b3e38ad9bd2)

**Comment by misterneo** [comment]
Hello @danharrin, Can you please answer and let me know if this is suitable for a PR ? Thank you.

**Reply by danharrin** [reply]
I don't think this should be part of the utility, as it can be important to test the `hidden()` condition separately to the user's toggle prefernece.

You can do this `->assertTableColumnExists('name', fn ($column) => $column->isToggledHiddenByDefault())`

**Reply by misterneo** [reply]
I didn’t notice the `Closure` parameter in `assertTableColumnExists`, but now that you've pointed it out.  it’s clear how it solves the problem. However, it could be a bit redundant if the check needs to be invoked multiple times. To avoid this, I followed this approach:

```php
use Closure;
use Filament\Tables\Testing\TestsColumns as BaseTestsColumns;

class TestsColumns extends BaseTestsColumns
{
    public function assertTableColumnIsToggledHiddenByDefault(): Closure
    {
        return function (string $name): static {
            /** @phpstan-ignore-next-line */
            $this->assertTableColumnExists($name);

            $column = $this->instance()->getTable()->getColumn($name);

            $livewireClass = $this->instance()::class;

            \Illuminate\Testing\Assert::assertTrue(
                $column->isToggledHiddenByDefault(),
                message: "Failed asserting that a table column with name [{$name}] is toggled hidden by default on the [{$livewireClass}] component.",
            );

            return $this;
        };
    }
}
```
```php
Testable::mixin(new TestsColumns);
```

**Reply by danharrin** [reply]
I do not want to add additional helpers when the `exists` one is present, there are too many configuration methods and having a testing helper for each is impractical and a task that will never end. You can add these aliases in your own project if you would like

**Reply by misterneo** [reply]
Yes, I definitely agree. The code snippet I shared is actually from my own project, which allows me to create any additional helper methods I may need.
